lab_1/main.py

```diff
@@ -30,7 +43,17 @@ def calculate_frequencies(tokens: list) -> dict or None:
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
-    pass
+    freq_dict = {}
+    if type(tokens) == list:
```

Comment:
There should be a check for the input arguments at the top of the function - after the documentation string. Remove nesting here and extract the check into a separate `if` block earlier.